sdk/clientcore/tools/annotation-processor/src/main/java/io/clientcore/tools/codegen/AnnotationProcessor.java

```diff
@@ -0,0 +1,196 @@
+import io.clientcore.tools.codegen.models.TemplateInput;
+import io.clientcore.tools.codegen.templating.TemplateProcessor;
+import io.clientcore.tools.codegen.utils.PathBuilder;
+import io.clientcore.core.annotation.ServiceInterface;
+import io.clientcore.core.http.annotation.BodyParam;
+import io.clientcore.core.http.annotation.HeaderParam;
+import io.clientcore.core.http.annotation.HostParam;
+import io.clientcore.core.http.annotation.HttpRequestInformation;
+import io.clientcore.core.http.annotation.PathParam;
+import io.clientcore.core.http.annotation.QueryParam;
+
+import io.clientcore.core.http.annotation.UnexpectedResponseExceptionDetail;
+import javax.annotation.processing.AbstractProcessor;
+import javax.annotation.processing.RoundEnvironment;
+import javax.annotation.processing.SupportedAnnotationTypes;
+import javax.annotation.processing.SupportedSourceVersion;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.TypeElement;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@SupportedAnnotationTypes("io.clientcore.core.annotation.*")
+@SupportedSourceVersion(SourceVersion.RELEASE_8)
+public class AnnotationProcessor extends AbstractProcessor {
+
+    @Override
+    public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+        // We iterate through each interface annotated with @ServiceInterface separately.
+        // This outer for-loop is not strictly necessary, as we only have one annotation that we care about
+        // (@ServiceInterface), but we'll leave it here for now
+        annotations.stream()
+                .map(roundEnv::getElementsAnnotatedWith)
+                .flatMap(Set::stream)
+                .filter(element -> element.getKind().isInterface())
+                .forEach(element -> {
+                    if (element.getAnnotation(ServiceInterface.class) != null) {
+                        this.processServiceInterface(element);
+                    }
+                });
+
+        return true;
+    }
+
+
+    private void processServiceInterface(Element serviceInterface) {
+        TemplateInput templateInput = new TemplateInput();
+
+        // work out some global details about this service interface
+        final String serviceInterfaceFQN = serviceInterface.asType().toString();
+
+        String packageName = null;
+        int lastDot = serviceInterfaceFQN.lastIndexOf('.');
+        if (lastDot > 0) {
+            packageName = serviceInterfaceFQN.substring(0, lastDot);
+        }
+
+        final String serviceInterfaceShortName = serviceInterfaceFQN.substring(lastDot + 1);
+        final String serviceInterfaceImplFQN = serviceInterfaceFQN + "Impl";
+        final String serviceInterfaceImplShortName = serviceInterfaceImplFQN.substring(lastDot + 1);
```

Comment:
```suggestion
        final String serviceInterfaceShortName = serviceInterfaceFQN.substring(lastDot + 1);
        final String serviceInterfaceImplShortName = serviceInterfaceShortName + "Impl";
```

No need for `serviceInterfaceImplFQN` as it just gets thrown away and we're repeating the logic to substring.